virttest/migration_template.py

```diff
@@ -598,6 +736,30 @@ def _set_vm_disk(self, cache="none"):
                 # TODO:other storage types
                 self.test.cancel("Only nfs storage is supported for now")
 
+    def _create_disk_image_on_dest(self):
+        """
+        Create disk image on dest host before migration
+        Used for live vm migration with disk copy
+
+        Note: 
+        This method doesn't handle the backing chain setup. So you need to setup
+        the disk image backing chain by yourself if --copy-storage-inc is used
+
+        """
+        logging.debug("Create disk image on dest host before migration")
+        if self.migrate_flags & (VIR_MIGRATE_NON_SHARED_DISK|VIR_MIGRATE_NON_SHARED_INC):
+            all_vm_disks = self.main_vm.get_blk_devices()
+            for disk in list(itervalues(all_vm_disks)):
+                disk_type = disk.get("type")
+                disk_path = disk.get("source")
+                disk_size = utils_misc.get_image_info(disk_path).get("vsize")
+                disk_format = utils_misc.get_image_info(disk_path).get("format")
```

Comment:
We might invoke get_image_info() once and then read the returned dict for 'vsize' and 'format'. That will be better.

Reply:
Updated